src/compiler/crystal/tools/formatter.cr

```diff
@@ -4515,8 +4515,15 @@ module Crystal
         value = @token.value.to_s.strip
         raw_after_comment_value = value[1..-1]
         after_comment_value = raw_after_comment_value.strip
-        if after_comment_value.starts_with?("=>")
-          value = "\# => #{after_comment_value[2..-1].strip}"
+        if formatted_result_expression = after_comment_value.lchop? "=>"
```

Comment:
Minor detail, but it's not formatted at the beginning, so I'd just call this variable `result_expression` (everywhere).
```suggestion
        if result_expression = after_comment_value.lchop? "=>"
```